### DISCUSSION TITLE
Mamba-based installation on linux fails --pkg_resources.DistributionNotFound: 'setuptools' by pynwb

### DISCUSSION THREAD

**Discussion by maedat** [talmolab/sleap#1467] [Help!]
Thank you for developing this wonderful software.

We installed SLEAP on Linux (Ubuntu 20.04.5 LTS) via following command
`mamba create -y -n sleap -c conda-forge -c nvidia -c sleap -c anaconda sleap=1.3.1`

Installation was completed successfully.
However, sleap-label and sleap-diagnostic command (after "conda activate sleap")
give following errors

`pkg_resources.DistributionNotFound: The 'setuptools' distribution was not found and is required by pynwb`

We confirmed the installation. But, 
'mamba install setuptools 
mamba install pynwb'
showed that the packages have been installed. 

This may be a problem on the pynwb side, but if there is any solution, please let me know.

**Comment by antortjim** [comment]
Maybe try running `mamba activate` instead of `conda activate`. AFAIK they don't do the same thing!

**Comment by roomrys** [comment]
Hi @maedat,

There is an open issue on this #1428, do you mind if we follow up there? I'll comment in both places, then close this discussion and see you in #1428!

## Questions for you
__Before you try reinstalling__, can you activate the broken SLEAP installation and run the commands
```bash
mamba list
pip freeze
```
and post the output for both.

__Is this a new installation or are you upgrading your version of SLEAP?__ I would expect that a completely new installation (never before installed SLEAP... or any other packages using conda) would not have this problem and would be a bit more concerned if it is the former.

## Possible Solution (let us know how it goes!)
Do you mind trying the command:
>I created and uploaded a new build for sleap 1.3.1 linux-64. To make sure you are pulling in the new build instead a cached one, do you mind trying the command:
>```bash
>mamba create -y -n sleap_test_0 -c conda-forge -c nvidia -c sleap -c anaconda sleap=1.3.1=py37_11
>```
>Then run
>```bash
>mamba activate sleap_test_0
>```
>and finally
>```bash
>sleap-label
>```

## What is different in `sleap=1.3.1=py37_11`?
I retried the Linux installation and ran into this error, so I pinned `pynwb <= 2.3.3` and rebuilt, but it was still happening. Another difference I noticed in dependencies (comparing to an already installed SLEAP 1.3.1 package) was the `tensorflow-hub` (and therefore `protobuf`) versions. So I pinned `tensorflow-hub <= 1.13.0` and rebuilt, which seemed to make it work for me. Some extra steps I also took were that I cleared my conda and pypi cache before creating the environment to ensure there weren't any local preferences taking place....
```
mamba clean -a
pip cache purge
```
If the reason it started working for me was the `mamba clean` or `pip cache purge`, then that indicates that the version ranges are too broad, but I still wanted to include it in case you had trouble with installing the specific build `sleap=1.3.1=py37_11`.

Thanks! See you in #1428,
Liezl
